content/essentials/signals.mdx

```diff
@@ -0,0 +1,105 @@
+
+```jsx
+import { createSignal } form "solid-js";
+```
+Upon its call, this function yields a pair of values: the first is a **getter **function, which is responsible for _accessing _the current value of the signal, and the second is a **setter **function, designed to _update _the signal's value:
+
+```jsx
+const [count, setCount] = createSignal(0);
+```
+
+In this example, `count` is the getter and `setCount`  is the setter function.
+
+<Aside>
+
+The syntax using `[` and `]` is called [array destructuring](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment). 
+This lets you extract values from the array. 
+Specifically, in the context of `createSignal` , there are two items being returned: the getter (first), and the setter (second).
+
+</Aside>
+
+Signals allow you to manage a variety of data types, including both objects and arrays. 
+This feature, combined with the fact Solid components do not own state, allows you to bring Signals in from anywhere in your application, removing the need for additional third-party state management libraries.
+
+## Accessing values
+
+To retrieve the value of a signal, you need to call the getter function:
+
+```jsx
+console.log(count()) // output: 0
+```
+
+When a signal is called within a [tracking scope](/essentials/intro-to-reactivity#tracking), it forms a dependency between the signal and the wrapping function. 
+In this context, if the value of the signal changes, it notifies its dependencies of this change. 
+As a result, anything that is subscribed to this signal by accessing it within a tracking scope, will be re-evaluated or re-run. 
+This ensures that anything dependent on this signal will always be in sync with the state represented by these signals.
+
+<Aside>
+
+A tracking scope is typically created by [`createEffect`](/reference/createEffect)  or [`createMemo`](/reference/createMemo), other Solid primitives. 
+Both functions subscribe to the signals accessed within them, establishing a dependency relationship.
+
+When a signal within this scope changes its value, these dependencies are identified, and the functions or computations relying on them are triggered to update.
+
+</Aside>
+
+## Updating values
+
+To update a signal's value, you just need to invoke the setter function returned by `createSignal`:
+
+```jsx
+setCount(count()+1)
+
+console.log(count()) // output: 1
+```
+
+When the setter is called, the value of the signal is automatically updated, triggering the required updates in the functions that are tracking that signal.
+
+## Reacting to changes
+
+A distinct feature of signals is their ability to keep a list of their dependents. 
+[Effects](/essentials/effects) and [memos](/essentials/memos) subscribe to signals, inherently becoming _dependent _on them.
+This ensures a streamlined updating process across all the dependent functions whenever the signal's value changes.
```

Comment:
Unclear what a "streamlined process" means, like I don't think we need to add adjectives to the process, we probably just need to say what happens here. Like now I'm thinking what a non-streamlined way to do this is.